# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: extensions/permissions/PermissionDelegateHandler.cpp
# Commit: 202c4f5c642b
# Full Hash: 202c4f5c642b8ec35e849c2144f935342d193fd7
# Author: Vincent Hilla <vhilla@mozilla.com>
# Date: 2023-12-05 21:29:44
# Regressor Bug: 1589554
# File Overlap Count: 1
# Description:
#   Bug 1589554 - Part 3: Screen Wake Lock permission integration. r=webidl,dom-core,smaug,pbz,edgar
#   
#   Depends on D189509
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D189510
# ==============================================================================

diff -r 82e03a404c2f -r 202c4f5c642b extensions/permissions/PermissionDelegateHandler.cpp
--- a/extensions/permissions/PermissionDelegateHandler.cpp	Tue Dec 05 15:37:05 2023 +0000
+++ b/extensions/permissions/PermissionDelegateHandler.cpp	Tue Dec 05 15:37:06 2023 +0000
@@ -45,7 +45,8 @@
     {"microphone", u"microphone", DelegatePolicy::eDelegateUseFeaturePolicy},
     {"screen", u"display-capture", DelegatePolicy::eDelegateUseFeaturePolicy},
     {"xr", u"xr-spatial-tracking", DelegatePolicy::eDelegateUseFeaturePolicy},
-};
+    {"screen-wake-lock", u"screen-wake-lock",
+     DelegatePolicy::eDelegateUseFeaturePolicy}};
 
 static_assert(PermissionDelegateHandler::DELEGATED_PERMISSION_COUNT ==
                   (sizeof(sPermissionsMap) / sizeof(DelegateInfo)),